src/decorators/connect.ts

```diff
@@ -62,18 +65,36 @@ function updateConfig(target: Constructor<any>, config: Connection) {
   }
 }
 
-function construct(options: ConstructOptions) {
+function construct(options: ConnectedCreationOptions) {
   const {stateMapper, actionMapper}: Connection = Reflect.getOwnMetadata(componentConfigKey, options.proxy);
-  const target = Reflect.construct(options.target, options.constructorArgs, options.protoTarget);
+  const target = Reflect.construct(options.target, options.creationArgs, options.context);
   if (stateMapper || actionMapper) {
-    const stateProvider = getInjectable(StateProvider, options.constructorArgs);
+    const stateProvider = getInjectable(StateProvider, options.creationArgs);
     StateProvider.hook({stateProvider, target, stateMapper, actionMapper});
   }
   return target;
 }
 
-function addInjectorInjection(proxy: Constructor<any>, config: Connection) {
-  if (config.stateMapper) {
+function apply(options: ConnectedCreationOptions) {
+  const connection: Connection = Reflect.getOwnMetadata(componentConfigKey, options.proxy);
+  const target = Reflect.apply(options.target, options.context, options.creationArgs);
+  return connectTarget(target, connection, options);
+}
+
+function connectTarget(
```

Comment:
Maybe inline this function? It seems to be used only at one place and the parameter declaration is almost as long as the function itself

Reply:
Was supposed to be used twice. Now it is.